lib/smart_proxy_openscap/openscap_lib.rb

```diff
@@ -13,10 +13,31 @@
 require 'json'
 require 'proxy/error'
 require 'proxy/request'
+require 'smart_proxy_openscap/openscap_exception'
 
 module Proxy::OpenSCAP
   extend ::Proxy::Log
 
+  def self.get_policy_content(policy_id)
+    policy_store_dir = File.join(Proxy::OpenSCAP::Plugin.settings.contentdir, policy_id.to_s)
+    policy_scap_file = File.join(policy_store_dir, "#{policy_id}_scap_content.xml")
+    FileUtils.mkdir_p(policy_store_dir) # will fail silently if exists
+
+    return File.open(policy_scap_file, 'rb').read if File.file?(policy_scap_file)
+
+    foreman_request = Proxy::HttpRequest::ForemanRequest.new
+    policy_content_path = "/api/v2/compliance/policies/#{policy_id}/content"
+    req = foreman_request.request_factory.create_get(policy_content_path)
+    response = foreman_request.send_request(req)
+    unless response.is_a? Net::HTTPSuccess
+      raise OpenSCAPException.new(response)
+    end
+    File.open(policy_scap_file, 'wb') do |file|
```

Comment:
This has a concurrency issue If multiple clients attempt to retrieve a non-local policy. https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/http_downloads.rb#L6 is currently used in smart-proxy to resolve this issue (used here: https://github.com/theforeman/smart-proxy/blob/develop/modules/tftp/server.rb#L106 and https://github.com/theforeman/smart-proxy/blob/develop/modules/puppet_proxy/puppet_cache.rb#L72)